sunpy/coordinates/wcs_utils.py

```diff
@@ -74,6 +142,19 @@ def solar_wcs_frame_mapping(wcs):
                              attrs[2] * u.m,
                              **kwargs)
 
+    # Read the observer out of obsgeo for ground based observers
+    if observer is None:
+        try:
+            observer = obsgeo_to_frame(wcs.wcs.obsgeo, dateobs)
+            observer = SkyCoord(observer, rsun=rsun)
+        except ValueError as e:
+            # The helper function assumes you know the obsgeo coords you are
+            # parsing are good, we are not sure, so catch the error.
+
+            # This approach could lead to an invalid observer (i.e. one of the
+            # coords being NaN), but only if the WCS has been constructed like that.
+            log.debug(f"Could not parse obsgeo coordinates from WCS:\n{e}")
```

Comment:
Will this emit a message even if there is no observer defined on the WCS? Perhaps an additional check should be added to L146 to see if any `wcs.wcs.obsgeo` values are not their default values?

Reply:
It will. The issue is that the detection logic is frail and a little involved so I didn't want to duplicate it here. I added this log only as a way of debugging it if you think it should be working and isn't. I guess just adding a simple `is not None` would wipe out most of the cases where you don't want to hit that. That being said there are cases in the sunpy test suite where it's not None it's all NaNs. So there are just way too many cases to catch and I didn't want to replicate the checks in the upstream function in here.